packages/salesforcedx-vscode-core/src/commands/isvdebugging/bootstrapCmd.ts

```diff
@@ -64,52 +78,278 @@ export class IsvDebugBootstrapExecutor extends SfdxCommandletExecutor<{}> {
+        nls.localize(
+          'isv_debug_bootstrap_step7_convert_package_source',
+          packageName
+        )
+      )
+      .withArg('force:mdapi:convert')
+      .withFlag(
+        '-r',
+        path.join(this.relativeMetdataTempPath, 'packages', packageName)
+      )
+      .withFlag('-d', path.join('packages', packageName))
+      .build();
+  }
+
+  public parsePackageInstalledListJson(packagesJson: string): string[] {
+    const packagesData = JSON.parse(packagesJson);
+    return packagesData.result.map((entry: any) => entry.SubscriberPackageName);
+  }
+
+  public async execute(
+    response: ContinueResponse<IsvDebugBootstrapConfig>
+  ): Promise<void> {
     const cancellationTokenSource = new vscode.CancellationTokenSource();
     const cancellationToken = cancellationTokenSource.token;
 
     const projectParentPath = response.data.projectUri;
     const projectPath = path.join(projectParentPath, response.data.projectName);
+    const projectMetadataTempPath = path.join(
+      projectPath,
+      this.relativeMetdataTempPath
+    );
+    const apexRetrievePackageXmlPath = path.join(
+      projectPath,
+      this.relativeApexPackageXmlPath
+    );
 
-    const createProjectExecution = new CliCommandExecutor(
+    // 1: create project
+    await this.executeCommand(
       this.buildCreateProjectCommand(response.data),
-      {
-        cwd: projectParentPath
-      }
-    ).execute(cancellationToken);
-
-    createProjectExecution.processExitSubject.subscribe(async data => {
-      if (data != undefined && data.toString() === '0') {
-        const configureProjectExecution = new CliCommandExecutor(
-          this.buildConfigureProjectCommand(response.data),
-          {
-            cwd: projectPath
-          }
-        ).execute(cancellationToken);
-
-        configureProjectExecution.processExitSubject.subscribe(async data2 => {
-          if (data2 != undefined && data2.toString() === '0') {
-            // last step is open the folder
-            await vscode.commands.executeCommand(
-              'vscode.openFolder',
-              vscode.Uri.parse(projectPath)
-            );
-          }
-        });
+      { cwd: projectParentPath },
+      cancellationTokenSource,
+      cancellationToken
+    );
 
-        this.attachExecution(
-          configureProjectExecution,
-          cancellationTokenSource,
-          cancellationToken
-        );
-      }
-    });
+    // 2: configure project
+    await this.executeCommand(
+      this.buildConfigureProjectCommand(response.data),
+      { cwd: projectPath },
+      cancellationTokenSource,
+      cancellationToken
+    );
+
+    // 3a: create package.xml for downloading org apex
+    try {
+      this.mkdirSyncRecursive(projectMetadataTempPath);
+      fs.writeFileSync(
+        apexRetrievePackageXmlPath,
+        `<?xml version="1.0" encoding="UTF-8"?>
+<Package xmlns="http://soap.sforce.com/2006/04/metadata">
+  <version>41.0</version>
```

Comment:
How will this get updated as releases go out? Do we want to prompt the user to specify a version?

Reply:
Good catch. I haven't thought about this yet. I'll try without the `version` element. This may trigger the CLI to use whatever version the CLI is defaulting too.